content/docs/guides/webassembly/wasi.md

```diff
@@ -0,0 +1,30 @@
+---
+title: "Using WASI"
+weight: 3
+description: |
+  How to use TinyGo with the WebAssembly System Interface (WASI).
+---
+
+TinyGo is very useful for compiling programs for use on servers and other edge devices (WASI).
+
+TinyGo programs can run in Fastly Compute@Edge (https://developer.fastly.com/learning/compute/go/), Fermyon Spin (https://developer.fermyon.com/spin/go-components), wazero (https://wazero.io/languages/tinygo/) and many other WebAssembly runtimes.
+
+Here is a small TinyGo program for use within a WASI host application:
+
+```go
+package main
+
+//export add
+func add(x, y uint32) uint32 {
+	return x + y
+}
```

Comment:
Did you test this? Because I suspect it also needs a `//go:wasm-module` (at least until a hypothetical `//go:wasmexport` is defined).

Reply:
Umm, I did not test it yet.

Reply:
What should the module name be here?

```//go:wasm-module ???```

Reply:
That's up to you. It's the name that must be given when importing the function on the other side. (The usual default is `env`, I think).

Reply:
Added.